pennylane/grouping/group_observables.py

```diff
@@ -17,10 +17,10 @@
 
 from pennylane.wires import Wires
 from pennylane.grouping.utils import (
-    convert_observables_to_binary_matrix,
+    observables_to_binary_matrix,
```

Comment:
Sumarry of the renaming:
    - `diagonalize_qwc_grouping` -> `diagonalize_qwc_pauli_words`
    - `obtain_qwc_post_rotations_and_diagonalized_groupings` -> `diagonalize_qwc_groupings`
    - `convert_observables_to_binary_matrix` -> `observables_to_binary_matrix`
    - `get_qwc_complement_adj_matrix` -> `qwc_complement_adj_matrix`

Reply:
Thanks @antalszava!

@co9olguy, for context as to why these were renamed: they include redundant terms (`get`, `obtain`, `convert`), and are too long - they were causing excessive horizontal scrolling and word wrapping in the documentation. The latter point wasn't evident until this PR, when they were added to the API documentation.